kernel/data-pipeline/dialect/opengauss/src/main/java/org/apache/shardingsphere/data/pipeline/opengauss/ingest/OpenGaussWALDumper.java

```diff
@@ -99,6 +113,23 @@ private PgConnection getReplicationConnectionUnwrap() throws SQLException {
         return logicalReplication.createConnection((StandardPipelineDataSourceConfiguration) dumperConfig.getDataSourceConfig()).unwrap(PgConnection.class);
     }
     
+    private void processEventWithXid(final AbstractWALEvent event) {
+        if (event instanceof AbstractRowEvent) {
+            rowEvents.add((AbstractRowEvent) event);
+            return;
+        }
+        if (event instanceof BeginXidEvent) {
+            rowEvents.clear();
+        }
+        if (event instanceof CommitXidEvent) {
+            for (AbstractRowEvent each : rowEvents) {
+                // TODO if multiple threads are pushing data in a channel at the same time, the order of the data will be incorrect
```

Comment:
Every `channel` instance could be used only by current OpenGaussWALDumper instance, it's the current logic. Then use merge-order on diferent channels.
To reuse more current code.